### ISSUE TITLE
Docker Failed to Initialize

### ISSUE THREAD

**Issue by JamesCohen-awin** [docker/for-win#907]
### Expected behavior
Docker Engine would start

### Actual behavior
Received an error message "Docker Failed to Initialize"

### Information
```
Version: 17.06.0-ce-win18 (12627)
Channel: stable
Sha1: d9b66511e0920fdb4a7ff88b8343f65b46fc07e0
Started on: 2017/07/10 09:39:00.255
Resources: C:\Program Files\Docker\Docker\Resources
OS: Windows 10 Pro
Edition: Professional
Id: 1607
Build: 14393
BuildLabName: 14393.1358.amd64fre.rs1_release.170602-2252
File: C:\Users\james.cohen.DOMAIN\AppData\Local\Docker\log.txt
CommandLine: "C:\Program Files\Docker\Docker\Docker for Windows.exe" 
You can send feedback, including this log file, at https://github.com/docker/for-win/issues
[09:39:00.696][GUI            ][Info   ] Starting...
[09:39:02.020][Tracking       ][Info   ] Crash report and usage statistics are enabled
[09:39:02.034][SegmentApi     ][Info   ] Usage statistic: appLaunched
[09:39:02.192][NamedPipeClient][Info   ] Sending Version()...
[09:39:32.411][GUI            ][Error  ] A task was canceled.
[10:25:14.444][ErrorReportWindow][Info   ] Open logs
```

Crash report id: `1431A2DE-2CC1-4A2D-B291-C723A6BD1056/2017-07-10_10-27-23`

### Steps to reproduce the behavior

Turn on machine. Wait for it to launch Docker via the startup.

**Comment by jasonbivins** [comment]
Hi @JamesCohen-awin   The logs only show what we're seeing here - that "A task was canceled"
Can you run this debug script and share the results?  This will check to make sure you meet system requirements and that they are all functional

https://github.com/Microsoft/Virtualization-Documentation/tree/live/windows-server-container-tools/Debug-ContainerHost

**Comment by DaleCulpin** [comment]
I'm getting this same issue.

Occasionally it will start up fine, but mostly it fails with this **'Failed to initialize'** message.

I'll try running the linked diagnostics script above when I get home tonight and post results.

**Comment by DaleCulpin** [comment]
Here is the Powershell output from running the diagnostic script on my Windows 10 Pro 64-bit machine;

`Checking for common problems
Describing Windows Version and Prerequisites
 [+] Is Windows 10 Anniversary Update or Windows Server 2016 47ms
 [+] Has KB3192366, KB3194496, or later installed if running Windows build 14393 12ms
 [+] Is not a build with blocking issues 14ms
 [-] Has 'Containers' feature installed 40.84s
   Expected: {Enabled}
   But was:  {Disabled}
   23:             (Get-WindowsOptionalFeature -Online -FeatureName Containers).State | Should Be "Enabled"
   at <ScriptBlock>, C:\Users\Pilmar\Desktop\Docker Diagnostic\docker_diagnostic.ps1: line 23
Describing Docker is installed
 [+] A Docker service is installed - 'Docker' or 'com.Docker.Service'  24ms
 [+] Service is running 13ms
 [+] Docker.exe is in path 2.03s
 [+] Docker is registered in the EventLog service 53ms
Describing User has permissions to use Docker daemon
 [+] docker.exe should not return access denied 21ms
Describing Windows container settings are correct
 [+] Do not have DisableVSmbOplock set to 1 20ms
 [+] Do not have zz values set 14ms
 [+] Do not have FDVDenyWriteAccess set to 1 19ms
Describing The right container base images are installed
 [-] At least one of 'microsoft/windowsservercore' or 'microsoft/nanoserver' should be installed 57ms
   ValidationMetadataException: The argument is null or empty. Provide an argument that is not null or empty, and then try the command again.
   ParameterBindingValidationException: Cannot validate argument on parameter 'Property'. The argument is null or empty. Provide an argument that is not null or empty, and then try the command again.
   at <ScriptBlock>, C:\Users\Pilmar\Desktop\Docker Diagnostic\docker_diagnostic.ps1: line 115
Describing Container network is created
 [+] At least one local container network is available 1.14s
 [-] At least one NAT, Transparent, or L2Bridge Network exists 83ms
   Expected {0} to be greater than {0}
   206:       $totalnets | Should BeGreaterThan 0
   at <ScriptBlock>, C:\Users\Pilmar\Desktop\Docker Diagnostic\docker_diagnostic.ps1: line 206
 [-] NAT Network's vSwitch is internal 52ms
   Expected: {Internal}
   But was:  {}
   211:       $switchType | Should Be "Internal"
   at <ScriptBlock>, C:\Users\Pilmar\Desktop\Docker Diagnostic\docker_diagnostic.ps1: line 211
 [+] A Windows NAT is configured if a Docker NAT network exists 33ms
 [-] Specified Network Gateway IP for NAT network is assigned to Host vNIC 18ms
   Expected: value to not be empty
   225:       $natGatewayIP | Should Not BeNullOrEmpty
   at <ScriptBlock>, C:\Users\Pilmar\Desktop\Docker Diagnostic\docker_diagnostic.ps1: line 225
 [-] NAT Network's internal prefix does not overlap with external IP' 19ms
   Expected {0} to be greater than {0}
   252:         $hostips.Count | Should BeGreaterThan 0
   at <ScriptBlock>, C:\Users\Pilmar\Desktop\Docker Diagnostic\docker_diagnostic.ps1: line 252`

The CSV log is empty though.

**Comment by DaleCulpin** [comment]
Checking for common problems
Describing Windows Version and Prerequisites
 [+] Is Windows 10 Anniversary Update or Windows Server 2016 3.85s
 [+] Has KB3192366, KB3194496, or later installed if running Windows build 14393 331ms
 [+] Is not a build with blocking issues 56ms
 [+] Has 'Containers' feature installed 41.86s
Describing Docker is installed
 [+] A Docker service is installed - 'Docker' or 'com.Docker.Service'  275ms
 [+] Service is running 18ms
 [+] Docker.exe is in path 2.07s
 [+] Docker is registered in the EventLog service 64ms
Describing User has permissions to use Docker daemon
 [+] docker.exe should not return access denied 54ms
Describing Windows container settings are correct
 [+] Do not have DisableVSmbOplock set to 1 34ms
 [+] Do not have zz values set 52ms
 [+] Do not have FDVDenyWriteAccess set to 1 28ms
Describing The right container base images are installed
 [-] At least one of 'microsoft/windowsservercore' or 'microsoft/nanoserver' should be installed 135ms
   ValidationMetadataException: The argument is null or empty. Provide an argument that is not null or empty, and then try the command again.
   ParameterBindingValidationException: Cannot validate argument on parameter 'Property'. The argument is null or empty. Provide an argument that is not null or empty, and then try the command again.
   at <ScriptBlock>, C:\Users\Pilmar\Desktop\Docker Diagnostic\docker_diagnostic.ps1: line 115
Describing Container network is created
 [+] At least one local container network is available 1.11s
 [-] At least one NAT, Transparent, or L2Bridge Network exists 84ms
   Expected {0} to be greater than {0}
   206:       $totalnets | Should BeGreaterThan 0
   at <ScriptBlock>, C:\Users\Pilmar\Desktop\Docker Diagnostic\docker_diagnostic.ps1: line 206
 [-] NAT Network's vSwitch is internal 49ms
   Expected: {Internal}
   But was:  {}
   211:       $switchType | Should Be "Internal"
   at <ScriptBlock>, C:\Users\Pilmar\Desktop\Docker Diagnostic\docker_diagnostic.ps1: line 211
 [+] A Windows NAT is configured if a Docker NAT network exists 332ms
 [-] Specified Network Gateway IP for NAT network is assigned to Host vNIC 90ms
   Expected: value to not be empty
   225:       $natGatewayIP | Should Not BeNullOrEmpty
   at <ScriptBlock>, C:\Users\Pilmar\Desktop\Docker Diagnostic\docker_diagnostic.ps1: line 225
 [-] NAT Network's internal prefix does not overlap with external IP' 34ms
   Expected {0} to be greater than {0}
   252:         $hostips.Count | Should BeGreaterThan 0
   at <ScriptBlock>, C:\Users\Pilmar\Desktop\Docker Diagnostic\docker_diagnostic.ps1: line 252
